lib/Dialect/FIRRTL/Transforms/GrandCentralSignalMappings.cpp

```diff
@@ -0,0 +1,255 @@
+//===- GrandCentralSignalMappings.cpp ---------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the GrandCentralSignalMappings pass.
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+#include "circt/Dialect/FIRRTL/FIRRTLAnnotations.h"
+#include "circt/Dialect/FIRRTL/FIRRTLOps.h"
+#include "circt/Dialect/FIRRTL/Passes.h"
+#include "circt/Dialect/SV/SVDialect.h"
+#include "mlir/IR/ImplicitLocOpBuilder.h"
+#include "llvm/ADT/StringExtras.h"
+#include "llvm/Support/Debug.h"
+
+#define DEBUG_TYPE "gct"
```

Comment:
Yeah good point. I originally thought about having this common as "gct" since we were talking about merging all the GCT passes at some point. But seeing the extent of these highly specific transformations it looks like we'd be better off keeping them in separate passes. At the moment we still present them as a `--grand-central` flag to the user, so the "gct" might be fine as an analogy?